File: backend/app/schemas/matching.py
"""Institution matching API schemas."""

from pydantic import BaseModel, Field


class InstitutionMatch(BaseModel):
    """A ranked institution match."""

    institution_id: int
    institution_name: str
    institution_type: str
    location: str
    matched_capabilities: list[str]
    match_score: float = Field(ge=0, le=100)


class ChallengeMatchingResponse(BaseModel):
    """Institution matching results for a challenge."""

    challenge_id: int
    challenge_title: str
    required_capabilities: list[str]
    matches: list[InstitutionMatch]

File: frontend/src/types/matching.ts
/**
 * Institution matching types — aligned with backend app/schemas/matching.py
 */

export interface InstitutionMatch {
  institution_id: number;
  institution_name: string;
  institution_type: string;
  location: string;
  matched_capabilities: string[];
  match_score: number; // 0 to 100
}

export interface ChallengeMatchingResponse {
  challenge_id: number;
  challenge_title: string;
  required_capabilities: string[];
  matches: InstitutionMatch[];
}
